### ISSUE TITLE
Wondering if it's easy to add non-made hands classification

### ISSUE THREAD

**Issue by lei-cao** [HenryRLee/PokerHandEvaluator#20]
Good work!

Wondering if it's easy to add non-made hands classification like these in Flopzilla,
 flush-draw, straight draw, gutshot, backdoor draw, overpair, top pair, middle pair, weak pair

**Comment by HenryRLee** [comment]
The project is originally designed for efficient hand strength evaluation, so analytics such as probability calculation is not part of the scope. Classification is easy if it is classifying the best hand, which has only at most 7462 possibilities. But those classifications you mentioned, would be another story.

It is still possible using the same algorithm, but the work is not trivial. Basically, we need to extend the hash table for 5-card and 6-card hands, and save these classification information in the hash table. Flush draw is an exception, which need to be determined in the stage checking whether it is a flush. Overpair, which requires context of other players, is not possible unfortunately. The hash table is huge, as you can see, so it will take some amount of work.

**Comment by HenryRLee** [comment]
I haven't decided to add this feature to the repository yet. But if you have a repository forked/ported from this one, and it is open-source, I can help you work on it.

**Comment by lei-cao** [comment]
hi @HenryRLee ,
Thank you for the information. I am writing my own small tool in JS for training myself. I am just using this javascript port https://github.com/thlorenz/phe I am not sure if @thlorenz accept a pr for it or not.

I also thought of adding it myself, realized that to generate the hash table, I need to have the truth of source as you mentioned you were using Cactus Kev's evaluator. I didn't found one for classification, then probably I need to write the classification testing first then generate the hash.

**Comment by HenryRLee** [comment]
You're right, there is no easy truth of source like Cactus Kev's evaluator. What I would do is: implement the classifier myself, calculate the classification for each hand, and put the value to the hash table.

Flush draw is to count the number of cards in each suit. Pairs is to count the number of cards in each rank. Straight draw might be the trickiest, but since there are only 10 possible straights, then we can iterate each straight candidate and check if the hand fits the straight draw (e.g. for the straight 2-6, we can count the distinct number of cards in the range [2, 6] and check if it equals to 4).

If this classifier satisfies your requirement, maybe you don't need the framework of PHEvaluator at all. Anyway, feel free to ask any questions if you are stuck or having troubles.
